packages/scope-manager/src/referencer/ClassVisitor.ts

```diff
@@ -27,19 +29,70 @@ class ClassVisitor extends Visitor {
     classVisitor.visitClass(node);
   }
 
-  visit(node: TSESTree.Node | null | undefined): void {
-    // make sure we only handle the nodes we are designed to handle
-    if (node && node.type in this) {
-      super.visit(node);
-    } else {
-      this.#referencer.visit(node);
-    }
+  protected AccessorProperty(node: TSESTree.AccessorProperty): void {
+    this.visitPropertyDefinition(node);
   }
 
   ///////////////////
   // Visit helpers //
   ///////////////////
 
+  protected ClassBody(node: TSESTree.ClassBody): void {
+    // this is here on purpose so that this visitor explicitly declares visitors
+    // for all nodes it cares about (see the instance visit method above)
+    this.visitChildren(node);
+  }
+
+  protected Identifier(node: TSESTree.Identifier): void {
+    this.#referencer.visit(node);
+  }
+
+  protected MethodDefinition(node: TSESTree.MethodDefinition): void {
+    this.visitMethod(node);
+  }
+
+  protected PrivateIdentifier(): void {
+    // intentionally skip
+  }
+
+  protected PropertyDefinition(node: TSESTree.PropertyDefinition): void {
+    this.visitPropertyDefinition(node);
+  }
+
+  protected StaticBlock(node: TSESTree.StaticBlock): void {
+    this.#referencer.scopeManager.nestClassStaticBlockScope(node);
+
+    node.body.forEach(b => this.visit(b));
+
+    this.#referencer.close(node);
+  }
+
+  protected TSAbstractAccessorProperty(
+    node: TSESTree.TSAbstractAccessorProperty,
+  ): void {
+    this.visitPropertyDefinition(node);
+  }
+
+  /////////////////////
+  // Visit selectors //
+  /////////////////////
```

Comment:
This sorting is wrong.
The `visit*` helpers used to all be grouped at the top.

I don't have a problem with them being at the end -- but the comments need to be repositioned correctly.